src/transforms/sample.rs

```diff
@@ -154,9 +159,11 @@ impl FunctionTransform for Sample {
         self.count = (self.count + 1) % self.rate;
 
         if num % self.rate == 0 {
-            event
-                .as_mut_log()
-                .insert("sample_rate", self.rate.to_string());
+            match event {
+                Event::Log(ref mut event) => event.insert("sample_rate", self.rate.to_string()),
+                Event::Trace(ref mut event) => event.insert("sample_rate", self.rate.to_string()),
```

Comment:
```
mismatched types
expected reference `&vector_core::event::LogEvent`
   found reference `&TraceEvent`
in the same arm, a binding must have the same type in all alternativesrustc[E0308](https://doc.rust-lang.org/error-index.html#E0308)
```

Reply:
ah I see it now

Reply:
I think technically you could do:

```
 Event::Log(event) | Event::Trace(LogEvent(event))
```

But I think what you have is better anyway since we are trying to make the fact that `Event::Trace` just wraps log event an implementation detail rather than something you can depend on.